### ISSUE TITLE
CS8602 vs CA1062: Generics unconstrained vs constrained / inconsistencies

### ISSUE THREAD

**Issue by LukasGelke** [dotnet/roslyn-analyzers#6480]
### Analyzer

**Diagnostic IDs**:
- [CA1062](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1062): `Validate arguments of public methods`
- CS8602: `Dereference of a possibly null reference`

### Analyzer source
**Visual Studio**: Version 17.4.4 (Latest as of writing)
**DotNet**:
```
> dotnet --list-sdks
	6.0.405 [C:\Program Files\dotnet\sdk]
	7.0.102 [C:\Program Files\dotnet\sdk]
``` 
(Latest as of writing)

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.0 (Latest as of writing)

### Describe the bug

The Diagnostics may in some capacity allude to the same underlying potential issue (a null reference). But have different checks.
In my case, only one of each is reported in certain cases

### Steps To Reproduce

1. create new C# Project (eg Visual Studio Template)
	- use `<TargetFrameworks>net6.0-windows;net471</TargetFrameworks>`
2. add PackageReference to `Microsoft.CodeAnalysis.NetAnalyzers`
3. make sure CA1062 and CS8602 are both enabled
4. try the following snippet:
```cs
namespace DesignLibrary
{
  public sealed class Test
  {
    public static void Do<T>(T input)
    {
      _ = input.GetType();
    }

    public static void Do2<T>(T input)
      where T : class
    {
      _ = input.GetType();
    }
  }
}
```

### Expected behavior
- 4x CA1062 -> one in each method, each TargetFrameworks
- 4x CS8602 -> one in each method, each TargetFrameworks

So both methods have both warnings, in both TargetFrameworks. Technically 8 warnings in total.

### Actual behavior
- only one CA1062 in `Do2`, and only reported for `net471`
- only two CS8602, both in in `Do`, reported for both `net471` and `net6.0-windows`

## Additional context

(none)
